convert2rhel/actions/pre_ponr_changes/transaction.py

```diff
@@ -46,26 +46,19 @@ def run(self):
             transaction_handler.run_transaction(
                 validate_transaction=True,
             )
-            # TODO: Handling SystemExit here as way to speedup exception
-            # handling and not refactor contents of the underlying function.
-            # Most of the issues raised during that function call should be
-            # handled inside the function, so, it's safe for now to only catch
-            # SystemExit here, and later, change it to something more suitable.
+        except exceptions.CriticalError as e:
+            self.set_result(
+                level="ERROR",
+                id=e.id,
+                title=e.title,
+                description=e.description,
+                diagnosis=e.diagnosis,
+                remediation=e.remediation,
+                variables=e.variables,
+            )
+
         except SystemExit as e:
```

Comment:
Do we need to catch SystemExit here still?

Reply:
I figured it's better safe than sorry

Reply:
Note that the Action framework catches this too: https://github.com/oamg/convert2rhel/blob/main/convert2rhel/actions/__init__.py#L568

My recommendation would be, if you know that you are raising SystemExit somewhere, then go ahead and leave code in the Action plugin which catches it and does something (it's an UNKNOWN_ERROR because we know about the exception but we aren't able to differentiate it from other things that are also raising SystemExit).  If you think you've ported everything that can raise SystemExit, then you can let the framework handle it (It's an UNEXPECTED_ERROR because the action plugin did not expect anything to raise that exception).

(The message for the UNEXPECTED_ERROR also tells the user to file a bug so that we can fix it to have a message [or fix a bug] in future releases.)

Reply:
Ahh gotcha, thanks for the pointer